scripts/container-e2e.sh

```
@@ -0,0 +1,87 @@
+# (this example declares no proto constraints, so there is no validation path to assert)
+bye=$(buf curl --protocol grpc --http2-prior-knowledge -d '{"name":"Ada"}' \
+  "http://localhost:$PORT/greeter.v1.GreeterService/SayGoodbye" 2>/dev/null)
+echo "$bye" | grep -q 'Goodbye, Ada!' \
+  && ok "SayGoodbye -> \"Goodbye, Ada!\"" || bad "SayGoodbye" "reply=$bye"
+
+# 8. SIGTERM as PID 1: graceful shutdown, clean exit code, inside the grace window
+start=$(date +%s)
+docker stop -t 30 "$NAME" >/dev/null 2>&1
+elapsed=$(( $(date +%s) - start ))
+exit_code=$(docker inspect -f '{{.State.ExitCode}}' "$NAME" 2>/dev/null)
+[ "$exit_code" = "0" ] && ok "SIGTERM -> exit 0 (${elapsed}s)" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)